4.analyze-data/notebooks/UMAP/nbconverted/0.UMAP.py

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# ## Extract UMAP embeddings for CFReT CP Features
+
+# In[1]:
+
+
+import glob
+import pathlib
+import pandas as pd
+import umap
+
+from pycytominer import feature_select
+from pycytominer.cyto_utils import infer_cp_features
+
+
+# In[2]:
+
+
+# Set constants
+umap_random_seed = 1234
+umap_n_components = 2
+
+output_dir = pathlib.Path("results")
+
+
+# In[3]:
+
+
+# Set input paths
+data_dir = pathlib.Path("..", "..", "..", "3.process-cfret-features", "data")
+
+# Select only the feature selected files
+file_suffix = "*sc_norm_fs_cellprofiler.csv.gz"
+
+# Obtain file paths for all feature selected plates
+fs_files = glob.glob(f"{data_dir}/{file_suffix}")
+fs_files
+
+
+# In[4]:
+
+
+# Load feature data into a dictionary, keyed on plate name
+cp_dfs = {x.split("/")[-1]: pd.read_csv(x) for x in fs_files}
+
+# Print out useful information about each dataset
+print(cp_dfs.keys())
+[cp_dfs[x].shape for x in cp_dfs]
+
+
+# In[5]:
+
+
+# Fit UMAP features per dataset and save
+for plate_name in cp_dfs:
+    # Make sure to reinitialize UMAP instance per plate
+    umap_fit = umap.UMAP(
+        random_state=umap_random_seed,
+        n_components=umap_n_components
+    )
+    
+    # Remove NA columns
+    cp_df = cp_dfs[plate_name]
+    cp_df = feature_select(
+        cp_df,
+        operation="drop_na_columns",
+        na_cutoff=0
+    )
```

Comment:
I am curious about the NA columns. Are these NA columns for features? I have not seen this before, only in the metadata when the regular expression did not work.